Lib/contextlib.py

```diff
@@ -126,6 +133,44 @@ def __exit__(self, type, value, traceback):
                     raise
 
 
+class _AsyncGeneratorContextManager(_GeneratorContextManagerBase):
+    """Helper for @asynccontextmanager."""
+
+    async def __aenter__(self):
+        try:
+            return await self.gen.__anext__()
+        except StopAsyncIteration:
+            raise RuntimeError("generator didn't yield") from None
+
+    async def __aexit__(self, type, value, traceback):
+        if type is None:
+            try:
+                await self.gen.__anext__()
+            except StopAsyncIteration:
+                return
+            else:
+                raise RuntimeError("generator didn't stop")
+        else:
+            if value is None:
+                value = type()
+            # See _GeneratorContextManager.__exit__ for comments on subtleties
+            # in this implementation
+            try:
+                await self.gen.athrow(type, value, traceback)
+                raise RuntimeError("generator didn't stop after throw()")
+            except StopAsyncIteration as exc:
+                return exc is not value
+            except RuntimeError as exc:
+                if exc is value:
+                    return False
+                if exc.__cause__ is value:
+                    return False
+                raise
+            except:
+                if sys.exc_info()[1] is not value:
+                    raise
```

Comment:
Hitting this line means adding a test case that replaces the thrown in exception with an entirely unrelated one that is neither `StopAsyncIteration` nor `RuntimeError`